src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyGraph.cs

```diff
@@ -429,6 +465,13 @@ public class DependencyGraph
             Dictionary<DependencyDetail, Build> dependencyCache =
                 new Dictionary<DependencyDetail, Build>(new DependencyDetailComparer());
 
+            EarlyBreakOnType breakOnType = options.EarlyBuildBreak.Type;
+            HashSet<string> breakOn = null;
```

Comment:
>breakOn [](start = 28, length = 7)

Probably I'm not seeing where is this used but looks like it is only instanced once and adding a single BuildBreak type. If we are just setting a single type, do we need the whole HashSet compared to just passing the type?

Reply:
I wonder if the HashSet should also be a part of the EarlyBreakOn class so that it doesn't feel like it's so disconnected. We only read the option value once and then use this pair of (enum, HashSet) everywhere.

EarlyBreakOn can maybe also inform if a break has happened, cleaning up some of the code where we check for the Type + hashSet state.

Reply:
@riarenas @jcagme I did not want the graph build options to be modifed by the builder itself, which is the reason for the copy of the inputs.

We could pass around an EarlyBreakOn class of some kind

Reply:
Yeah.. I think it'd be weird to modify an option after passing it in.

I keep thinking what I don't like about this approach and I think in the end it's that anything that uses the BreakEarly option will need to add their copy of the inputs, and add their own logic like adding `breakOnType != EarlyBreakOnType.None && breakOn.Count == 0`  whenever they want to check if they should break, and feels like functionality that should be somewhat provided by helpers.

(Of course might be overthinking this scenario)

Reply:
I don't see any new scenarios needing BreakOn. It really is just an optimization.  Any new versions of this will need to figure out exactly where to insert such checks.

This could be changed as part of a larger refactoring (to...say...an event based build algorithm), but I don't think this is in scope for now.

Reply:
Yeah, it's probably just overkill at this point. I think the current approach is fine.